sound/soc/sof/pcm.c

```diff
@@ -84,7 +84,9 @@ EXPORT_SYMBOL(snd_sof_pcm_period_elapsed);
 
 static int
 sof_pcm_setup_connected_widgets(struct snd_sof_dev *sdev, struct snd_soc_pcm_runtime *rtd,
-				struct snd_sof_pcm *spcm, int dir)
+				struct snd_sof_pcm *spcm,
+				struct snd_sof_platform_stream_params *platform_params,
+				struct snd_pcm_hw_params *params, int dir)
```

Comment:
nitpick: I would have the following order: sdev, rtd, spcm, params, platform_params, dir

But rtd and spcm can be derived from substream if we want to keep the parameter list in check...